app/ports/utilities/sort_by_version.py

```diff
@@ -1,13 +1,59 @@
-from distutils.version import LooseVersion
+import re
+from functools import cmp_to_key
 
 
-def cleaned_version(version):
-    return version.replace("_", ".").replace("-", ".")
+# Tries to apply the same version sorting algorithm as used by macports-base
+# https://github.com/macports/macports-base/blob/master/src/pextlib1.0/vercomp.c
+def version_compare(version1, version2):
+    # Split on all non-alphanumeric characters and delete any empty ('') elements
+    # that might occur due to consecutive non-alphanumerics
+    version1_segments = list(filter(None, re.split('[^a-zA-Z0-9]', version1)))
+    version2_segments = list(filter(None, re.split('[^a-zA-Z0-9]', version2)))
+
+    version1_len = len(version1_segments)
+    version2_len = len(version2_segments)
+
+    # Now traverse the two lists and compare segments at each index
+    i = 0
+    while i < version1_len and i < version2_len:
```

Comment:
An alternative way (no need to change anything) could have been
{code}
for i in range(min(version1_len, version2_len)):
{code}

Reply:
Hmm...thanks, that would have been a nicer way.